Sources/CodeEditTextView/STTextViewController.swift

```diff
@@ -342,3 +342,30 @@ public class STTextViewController: NSViewController, STTextViewDelegate, ThemeAt
         highlighter = nil
     }
 }
+
+class CEScrollView: NSScrollView {
+
+    override func mouseDown(with event: NSEvent) {
+
+        if let textView = self.documentView as? STTextView,
+            !textView.visibleRect.contains(event.locationInWindow) {
+            // If the `scrollView` was clicked, but the click did not happen within the `textView`,
+            // set cursor to the last index of the `textView`.
+
+            guard let provider = textView.textLayoutManager.textContentManager else {
+                return
+            }
+
+            let string = textView.string
+
+            let range = NSRange(string.endIndex..<string.endIndex, in: string)
```

Comment:
If you could instead use the code below for finding the range, we'll avoid possibly copying the textView's string here. Swift should be smart enough not to do so but just in case. It'll also get rid of the optional unwrap.
```
let endLocation = textView.textLayoutManager.documentRange.endLocation
let range = NSTextRange(location: endLocation)
```

Reply:
Great suggestion, thanks. This also avoids needing the `guard let` for the provider.